File: app/src/main/java/app/kamy/saatApp/domain/share/ShareDayPeriod.kt
package app.kamy.saatApp.domain.share

import java.util.Calendar
import java.util.Date

enum class ShareDayPeriod {
    MORNING,
    AFTERNOON,
    EVENING,
    NIGHT;

    companion object {
        fun forDate(date: Date = Date(), calendar: Calendar = Calendar.getInstance()): ShareDayPeriod {
            return when (calendar.apply { time = date }.get(Calendar.HOUR_OF_DAY)) {
                in 5..11 -> MORNING
                in 12..15 -> AFTERNOON
                in 16..19 -> EVENING
                else -> NIGHT
            }
        }
    }
}

File: iosApp/Saat/Features/Discovery/Enums/ShareDayPeriod.swift
//
//  ShareDayPeriod.swift
//  Saat
//
//  Created by Elmee on 25/04/2026.
//  Copyright © 2026 Elmee. All rights reserved.
//

import Foundation

enum ShareDayPeriod: Sendable {
    case morning
    case afternoon
    case evening
    case night

    static func forDate(_ date: Date, calendar: Calendar = .current) -> ShareDayPeriod {
        switch calendar.component(.hour, from: date) {
        case 5..<12: return .morning
        case 12..<16: return .afternoon
        case 16..<20: return .evening
        default: return .night
        }
    }
}
